lib/domain/contact_manager/contacts_manager.dart

```diff
@@ -27,40 +55,99 @@ class ContactsManager {
 
-  StreamSubscription? contactsStreamSubscription;
+  StreamController<ContactsManagerArguments> contactsStream =
+      StreamController<ContactsManagerArguments>.broadcast();
+
+  ContactsManager() {
+    FlutterContacts.addListener(_fetchPhonebookContacts);
+  }
 
-  void initialSynchronizeContacts() {
+  void initialSynchronizeContacts() async {
+    if (PlatformInfos.isMobile && !_isNotDisplayWarningAgainCacheMemory) {
+      await _handleRequestContactsPermission();
+    }
     if (!firstSynchronized) {
       _syncContactsState = SyncContactsState.synchronizing;
       _getAllContacts();
+      _fetchPhonebookContacts();
+    } else {
+      _handleContactsHasDataChange();
+    }
+  }
+
+  Future<void> _handleRequestContactsPermission() async {
+    final currentContactsPermissionStatus =
+        await permissionHandlerService.requestContactsPermissionActions();
+    if (currentContactsPermissionStatus == PermissionStatus.granted) {
+      _isNotDisplayWarningAgainCacheMemory = true;
+      _displayWaringContactsBannerState = DisplayWaringContactsBannerState.hide;
+    } else {
+      if (!_isNotDisplayWarningAgainCacheMemory) {
+        _displayWaringContactsBannerState =
+            DisplayWaringContactsBannerState.show;
+      }
     }
   }
 
   void _getAllContacts() {
-    contactsStreamSubscription = _getAllContactsInteractor
+    _getAllContactsInteractor
         .execute(limit: AppConfig.maxFetchContacts)
         .listen((event) {
       event.fold(
         (_) {},
         (success) {
           if (success is GetContactsSuccess) {
             _tomContacts = success.tomContacts.toSet();
-            _phonebookContacts = {};
-            _syncContactsState = SyncContactsState.synchronized;
+            _handleContactsHasDataChange();
             Logs().d(
               "ContactManagerMixin()::getAllContacts(): TomContacts: ${_tomContacts.length}",
             );
-            Logs().d(
-              "ContactManagerMixin()::getAllContacts(): PhonebookContacts: ${_phonebookContacts.length}",
-            );
           }
         },
       );
     });
   }
+
+  void _fetchPhonebookContacts() async {
+    _phonebookContactInteractor.execute().listen((contacts) {
+      _phonebookContacts = contacts.toSet();
+      _handleContactsHasDataChange();
+      _lookupContacts(contacts);
+    });
+  }
+
+  void _lookupContacts(List<Contact> contacts) {
+    _lookupMxidInteractor
+        .execute(
+      phonebookContacts: contacts,
+      lookupChunkSize: _lookupChunkSize,
+    )
+        .listen((contacts) {
+      _phonebookContacts = contacts.toSet();
+      _handleContactsHasDataChange();
+    });
+  }
+
+  void closeContactsWarningBanner() {
+    _displayWaringContactsBannerState = DisplayWaringContactsBannerState.hide;
+    _isNotDisplayWarningAgainCacheMemory = true;
+    _handleContactsHasDataChange();
+  }
+
+  void _handleContactsHasDataChange() {
+    _syncContactsState = SyncContactsState.synchronized;
```

Comment:
oh, I saw you set the state to `synchronized` in one more place: 
- after get tom contacts
- after fetch phonebook contacts
- after close contact banner
- after lookup phone contact
What is the `real` one?

Reply:
no more set state
state will get from Either